--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -296,6 +_,10 @@
         net.minecraft.world.level.block.state.BlockBehaviour.BlockStateBase.canvas$initPathTypeCache();
         // Canvas end - pre start server config hook
         // Paper end - initialize global and world-defaults configuration
+        // TenseiMC start - initialize global and world-defaults configuration
+        this.tenseiHandlers.initializeGlobalConfiguration(this.registryAccess());
+        this.tenseiHandlers.initializeWorldDefaultsConfiguration(this.registryAccess());
+        // TenseiMC end - initialize global and world-defaults configuration
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
