From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Fri, 1 May 2026 13:31:18 +0200
Subject: [PATCH] Beacon Activation Range Configurable


diff --git a/net/minecraft/world/level/block/entity/BeaconBlockEntity.java b/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
index 54f29e7cbe425d2b7fb7dd967b637bb7b4b1ba52..48722d13af7b882d8897b1d1812dd75540582c79 100644
--- a/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
@@ -142,6 +142,15 @@ public class BeaconBlockEntity extends BlockEntity implements MenuProvider, Name
 
     public double getEffectRange() {
         if (this.effectRange < 0) {
+            if (this.getLevel() != null) {
+                return switch (this.levels) {
+                    case 1 -> this.getLevel().tensei.blocks.beacon.levelOne;
+                    case 2 -> this.getLevel().tensei.blocks.beacon.levelTwo;
+                    case 3 -> this.getLevel().tensei.blocks.beacon.levelThree;
+                    case 4 -> this.getLevel().tensei.blocks.beacon.levelFour;
+                    default -> this.levels * 10 + 10;
+                };
+            }
             return this.levels * 10 + 10;
         } else {
             return effectRange;
