From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Fri, 1 May 2026 13:31:18 +0200
Subject: [PATCH] Beacon Activation Range Configurable


diff --git a/net/minecraft/world/level/block/entity/BeaconBlockEntity.java b/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
index c2cf8df9bec98d372637b8c1e4c90444455f437a..8e9cf6bf85b7b0d6883d4288d282dc65bab8abd0 100644
--- a/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
@@ -138,6 +138,15 @@ public class BeaconBlockEntity extends BlockEntity implements MenuProvider, Name
 
     public double getEffectRange() {
         if (this.effectRange < 0) {
+            if (this.getLevel() != null) {
+                return switch (this.levels) {
+                    case 1 -> this.getLevel().tensei.blocks.beacon.levelOne;
+                    case 2 -> this.getLevel().tensei.blocks.beacon.levelTwo;
+                    case 3 -> this.getLevel().tensei.blocks.beacon.levelThree;
+                    case 4 -> this.getLevel().tensei.blocks.beacon.levelFour;
+                    default -> this.levels * 10 + 10;
+                };
+            }
             return this.levels * 10 + 10;
         } else {
             return effectRange;
