From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Fri, 1 May 2026 13:43:13 +0200
Subject: [PATCH] feat: make Bee hardcoded values configurable


diff --git a/net/minecraft/world/entity/animal/bee/Bee.java b/net/minecraft/world/entity/animal/bee/Bee.java
index a420b668a05379cd95e1377e9c6d63a1bbbc169d..7576740b0a8eeea88a57b6199d60c44d8cec1b4e 100644
--- a/net/minecraft/world/entity/animal/bee/Bee.java
+++ b/net/minecraft/world/entity/animal/bee/Bee.java
@@ -256,11 +256,12 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
             EnchantmentHelper.doPostAttackEffects(level, target, damageSource);
             if (target instanceof LivingEntity livingTarget) {
                 livingTarget.setStingerCount(livingTarget.getStingerCount() + 1);
+                var beeCfg = this.level().tensei.mobs.bee; // TenseiMC
                 int poisonTime = 0;
                 if (this.level().getDifficulty() == Difficulty.NORMAL) {
-                    poisonTime = 10;
+                    poisonTime = beeCfg.poisonSecondsNormal; // TenseiMC
                 } else if (this.level().getDifficulty() == Difficulty.HARD) {
-                    poisonTime = 18;
+                    poisonTime = beeCfg.poisonSecondsHard; // TenseiMC
                 }
 
                 if (poisonTime > 0) {
@@ -279,7 +280,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
     @Override
     public void tick() {
         super.tick();
-        if (this.hasNectar() && this.getCropsGrownSincePollination() < 10 && this.random.nextFloat() < 0.05F) {
+        if (this.hasNectar() && this.getCropsGrownSincePollination() < this.level().tensei.mobs.bee.maxCropsGrowable && this.random.nextFloat() < 0.05F) {
             for (int i = 0; i < this.random.nextInt(2) + 1; i++) {
                 this.spawnFluidParticle(
                     this.level(), this.getX() - 0.3F, this.getX() + 0.3F, this.getZ() - 0.3F, this.getZ() + 0.3F, this.getY(0.5), ParticleTypes.FALLING_NECTAR
@@ -345,30 +346,35 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
     }
 
     private boolean isTiredOfLookingForNectar() {
-        return this.ticksWithoutNectarSinceExitingHive > 3600;
+        return this.ticksWithoutNectarSinceExitingHive > this.level().tensei.mobs.bee.ticksWithoutNectarBeforeGoingHome; // TenseiMC
     }
 
     private void dropHive() {
         this.hivePos = null;
-        this.remainingCooldownBeforeLocatingNewHive = 200;
+        this.remainingCooldownBeforeLocatingNewHive = this.level().tensei.mobs.bee.cooldownBeforeLocatingNewHive; // TenseiMC
     }
 
     private void dropFlower() {
+        var beeCfg = this.level().tensei.mobs.bee; // TenseiMC
         this.savedFlowerPos = null;
-        this.remainingCooldownBeforeLocatingNewFlower = Mth.nextInt(this.random, 20, 60);
+        this.remainingCooldownBeforeLocatingNewFlower = Mth.nextInt(this.random, beeCfg.minFindFlowerRetryCooldown, beeCfg.maxFindFlowerRetryCooldown); // TenseiMC
     }
 
     private boolean wantsToEnterHive() {
         if (this.stayOutOfHiveCountdown <= 0 && !this.beePollinateGoal.isPollinating() && !this.hasStung() && this.getTarget() == null) {
             boolean wantsToEnterHive = this.hasNectar()
                 || this.isTiredOfLookingForNectar()
-                || this.level().environmentAttributes().getValue(EnvironmentAttributes.BEES_STAY_IN_HIVE, this.position());
+                || this.level().environmentAttributes().getValue(EnvironmentAttributes.BEES_STAY_IN_HIVE, this.position())|| isNightOrRaining(this.level()); // TenseiMC
             return wantsToEnterHive && !this.isHiveNearFire();
         } else {
             return false;
         }
     }
 
+    public static boolean isNightOrRaining(Level level) {
+        return level.dimensionType().hasSkyLight() && (level.isDarkOutside() && !level.tensei.mobs.bee.workAtNight || level.isRaining() && !level.tensei.mobs.bee.workInRain);
+    }
+
     public void setStayOutOfHiveCountdown(final int ticks) {
         this.stayOutOfHiveCountdown = ticks;
     }
@@ -388,6 +394,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
     @Override
     protected void customServerAiStep(final ServerLevel level) {
+        var beeCfg = this.level().tensei.mobs.bee; // TenseiMC
         boolean hasStung = this.hasStung();
         if (this.isInWater()) {
             this.underWaterTicks++;
@@ -395,13 +402,13 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
             this.underWaterTicks = 0;
         }
 
-        if (this.underWaterTicks > 20) {
+        if (this.underWaterTicks > beeCfg.underWaterTicksBeforeDamage) { // TenseiMC
             this.hurtServer(level, this.damageSources().drown(), 1.0F);
         }
 
         if (hasStung) {
             this.timeSinceSting++;
-            if (this.timeSinceSting % 5 == 0 && this.random.nextInt(Mth.clamp(1200 - this.timeSinceSting, 1, 1200)) == 0) {
+            if (this.timeSinceSting % 5 == 0 && this.random.nextInt(Mth.clamp(beeCfg.stingDeathCountdown - this.timeSinceSting, 1, beeCfg.stingDeathCountdown)) == 0) { // TenseiMC
                 this.hurtServer(level, this.damageSources().generic(), this.getHealth());
             }
         }
@@ -444,7 +451,8 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
     @Override
     public void startPersistentAngerTimer() {
-        this.setTimeToRemainAngry(PERSISTENT_ANGER_TIME.sample(this.random));
+        var beeCfg = this.level().tensei.mobs.bee; // TenseiMC
+        this.setTimeToRemainAngry(Mth.nextInt(this.random, beeCfg.angerTimeMin * 20, beeCfg.angerTimeMax * 20)); // TenseiMC
     }
 
     private boolean doesHiveHaveSpace(final BlockPos hivePos) {
@@ -495,7 +503,8 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
                 this.remainingCooldownBeforeLocatingNewFlower--;
             }
 
-            boolean shouldRoll = this.isAngry() && !this.hasStung() && this.getTarget() != null && this.getTarget().distanceToSqr(this) < 4.0;
+            int minAttackDist = this.level().tensei.mobs.bee.minAttackDistance; // TenseiMC
+            boolean shouldRoll = this.isAngry() && !this.hasStung() && this.getTarget() != null && this.getTarget().distanceToSqr(this) < (double)(minAttackDist * minAttackDist); // TenseiMC
             this.setRolling(shouldRoll);
             if (this.tickCount % 20 == 0 && !this.isHiveValid()) {
                 this.hivePos = null;
@@ -549,7 +558,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
     }
 
     private boolean isTooFarAway(final BlockPos targetPos) {
-        return !this.closerThan(targetPos, 48);
+        return !this.closerThan(targetPos, this.level().tensei.mobs.bee.tooFarDistance); // TenseiMC
     }
 
     private void setFlag(final int flag, final boolean value) {
@@ -772,7 +781,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
     private class BeeEnterHiveGoal extends Bee.BaseBeeGoal {
         @Override
         public boolean canBeeUse() {
-            if (Bee.this.hivePos != null && Bee.this.wantsToEnterHive() && Bee.this.hivePos.closerToCenterThan(Bee.this.position(), 2.0)) {
+            if (Bee.this.hivePos != null && Bee.this.wantsToEnterHive() && Bee.this.hivePos.closerToCenterThan(Bee.this.position(), Bee.this.level().tensei.mobs.bee.hiveCloseEnoughDistance)) { // TenseiMC
                 BeehiveBlockEntity beehiveBlockEntity = Bee.this.getBeehiveBlockEntity();
                 if (beehiveBlockEntity != null) {
                     if (!beehiveBlockEntity.isFull()) {
@@ -851,12 +860,12 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
         @Override
         public void tick() {
-            if (Bee.this.hivePos != null) {
+            if (Bee.this.hivePos != null) { var beeCfg = Bee.this.level().tensei.mobs.bee; // TenseiMC
                 this.travellingTicks++;
-                if (this.travellingTicks > this.adjustedTickDelay(2400)) {
+                if (this.travellingTicks > this.adjustedTickDelay(beeCfg.maxTravellingTicksToHive)) { // TenseiMC
                     this.dropAndBlacklistHive();
                 } else if (!Bee.this.navigation.isInProgress()) {
-                    if (!Bee.this.closerThan(Bee.this.hivePos, 16)) {
+                    if (!Bee.this.closerThan(Bee.this.hivePos, beeCfg.pathfindToHiveWhenCloserThan)) { // TenseiMC
                         if (Bee.this.isTooFarAway(Bee.this.hivePos)) {
                             Bee.this.dropHive();
                         } else {
@@ -868,7 +877,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
                             this.dropAndBlacklistHive();
                         } else if (this.lastPath != null && Bee.this.navigation.getPath().sameAs(this.lastPath)) {
                             this.ticksStuck++;
-                            if (this.ticksStuck > 60) {
+                            if (this.ticksStuck > beeCfg.ticksBeforeHiveDrop) { // TenseiMC
                                 Bee.this.dropHive();
                                 this.ticksStuck = 0;
                             }
@@ -894,7 +903,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         private void blacklistTarget(final BlockPos targetPos) {
             this.blacklistedTargets.add(targetPos);
 
-            while (this.blacklistedTargets.size() > 3) {
+            while (this.blacklistedTargets.size() > Bee.this.level().tensei.mobs.bee.maxBlacklistedHives) { // TenseiMC
                 this.blacklistedTargets.remove(0);
             }
         }
@@ -912,7 +921,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         }
 
         private boolean hasReachedTarget(final BlockPos targetPos) {
-            if (Bee.this.closerThan(targetPos, 2)) {
+            if (Bee.this.closerThan(targetPos, Bee.this.level().tensei.mobs.bee.hiveCloseEnoughDistance)) { // TenseiMC
                 return true;
             }
 
@@ -961,7 +970,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         public void tick() {
             if (Bee.this.savedFlowerPos != null) {
                 this.travellingTicks++;
-                if (this.travellingTicks > this.adjustedTickDelay(2400)) {
+                if (this.travellingTicks > this.adjustedTickDelay(Bee.this.level().tensei.mobs.bee.maxTravellingTicksToFlower)) { // TenseiMC
                     Bee.this.dropFlower();
                 } else if (!Bee.this.navigation.isInProgress()) {
                     if (Bee.this.isTooFarAway(Bee.this.savedFlowerPos)) {
@@ -974,7 +983,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         }
 
         private boolean wantsToGoToKnownFlower() {
-            return Bee.this.ticksWithoutNectarSinceExitingHive > 600;
+            return Bee.this.ticksWithoutNectarSinceExitingHive > Bee.this.level().tensei.mobs.bee.ticksBeforeGoingToKnownFlower; // TenseiMC
         }
     }
 
@@ -983,7 +992,8 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
         @Override
         public boolean canBeeUse() {
-            return Bee.this.getCropsGrownSincePollination() < 10 && !(Bee.this.random.nextFloat() < 0.3F) && Bee.this.hasNectar() && Bee.this.isHiveValid();
+            var beeCfg = Bee.this.level().tensei.mobs.bee; // TenseiMC
+            return Bee.this.getCropsGrownSincePollination() <  beeCfg.maxCropsGrowable && !(Bee.this.random.nextFloat() < beeCfg.cropGrowSkipChance) && Bee.this.hasNectar() && Bee.this.isHiveValid(); // TenseiMC
         }
 
         @Override
@@ -993,7 +1003,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
         @Override
         public void tick() {
-            if (Bee.this.random.nextInt(this.adjustedTickDelay(30)) == 0) {
+            if (Bee.this.random.nextInt(this.adjustedTickDelay(Bee.this.level().tensei.mobs.bee.cropGrowChance)) == 0) { // TenseiMC
                 for (int i = 1; i <= 2; i++) {
                     BlockPos belowPos = Bee.this.blockPosition().below(i);
                     BlockState belowState = Bee.this.level().getBlockState(belowPos);
@@ -1064,7 +1074,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
         @Override
         public void start() {
-            Bee.this.remainingCooldownBeforeLocatingNewHive = 200;
+            Bee.this.remainingCooldownBeforeLocatingNewHive = Bee.this.level().tensei.mobs.bee.cooldownBeforeLocatingNewHive; // TenseiMC
             List<BlockPos> hivesWithSpace = this.findNearbyHivesWithSpace();
             if (!hivesWithSpace.isEmpty()) {
                 for (BlockPos posToCheck : hivesWithSpace) {
@@ -1082,7 +1092,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         private List<BlockPos> findNearbyHivesWithSpace() {
             BlockPos beePos = Bee.this.blockPosition();
             PoiManager poiManager = ((ServerLevel)Bee.this.level()).getPoiManager();
-            Stream<PoiRecord> nearbyHives = poiManager.getInRange(p -> p.is(PoiTypeTags.BEE_HOME), beePos, Bee.HIVE_SEARCH_DISTANCE, PoiManager.Occupancy.ANY);
+            Stream<PoiRecord> nearbyHives = poiManager.getInRange(p -> p.is(PoiTypeTags.BEE_HOME), beePos, Bee.this.level().tensei.mobs.bee.hiveSearchDistance, PoiManager.Occupancy.ANY);
             return nearbyHives.map(PoiRecord::getPos)
                 .filter(Bee.this::doesHiveHaveSpace)
                 .sorted(Comparator.comparingDouble(pos -> pos.distSqr(beePos)))
@@ -1144,7 +1154,8 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
                         .moveTo(Bee.this.savedFlowerPos.getX() + 0.5, Bee.this.savedFlowerPos.getY() + 0.5, Bee.this.savedFlowerPos.getZ() + 0.5, 1.2F);
                     return true;
                 } else {
-                    Bee.this.remainingCooldownBeforeLocatingNewFlower = Mth.nextInt(Bee.this.random, 20, 60);
+                    var beeCfg = Bee.this.level().tensei.mobs.bee; // TenseiMC
+                    Bee.this.remainingCooldownBeforeLocatingNewFlower = Mth.nextInt(Bee.this.random, beeCfg.minFindFlowerRetryCooldown, beeCfg.maxFindFlowerRetryCooldown); // TenseiMC
                     return false;
                 }
             }
@@ -1159,7 +1170,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         }
 
         private boolean hasPollinatedLongEnough() {
-            return this.successfulPollinatingTicks > 400;
+            return this.successfulPollinatingTicks > Bee.this.level().tensei.mobs.bee.minPollinationTicks; // TenseiMC
         }
 
         private boolean isPollinating() {
@@ -1187,7 +1198,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
             this.pollinating = false;
             Bee.this.navigation.stop();
-            Bee.this.remainingCooldownBeforeLocatingNewFlower = 200;
+            Bee.this.remainingCooldownBeforeLocatingNewFlower = Bee.this.level().tensei.mobs.bee.cooldownBeforeLocatingNewFlower; // TenseiMC
         }
 
         @Override
@@ -1197,12 +1208,12 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
         @Override
         public void tick() {
-            if (Bee.this.hasSavedFlowerPos()) {
+            if (Bee.this.hasSavedFlowerPos()) { var beeCfg = Bee.this.level().tensei.mobs.bee; // TenseiMC
                 this.pollinatingTicks++;
-                if (this.pollinatingTicks > 600) {
+                if (this.pollinatingTicks > beeCfg.maxPollinationTicks) { // TenseiMC
                     Bee.this.dropFlower();
                     this.pollinating = false;
-                    Bee.this.remainingCooldownBeforeLocatingNewFlower = 200;
+                    Bee.this.remainingCooldownBeforeLocatingNewFlower = beeCfg.cooldownBeforeLocatingNewFlower; // TenseiMC
                 } else if (Bee.this.savedFlowerPos != null) { // Paper - add null check since API can manipulate this
                     Vec3 flowerPos = Vec3.atBottomCenterOf(Bee.this.savedFlowerPos).add(0.0, 0.6F, 0.0);
                     if (flowerPos.distanceTo(Bee.this.position()) > 1.0) {
@@ -1215,7 +1226,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
                         boolean arrivedAtHoverPos = Bee.this.position().distanceTo(this.hoverPos) <= 0.1;
                         boolean shouldSetWantedPos = true;
-                        if (!arrivedAtHoverPos && this.pollinatingTicks > 600) {
+                        if (!arrivedAtHoverPos && this.pollinatingTicks > beeCfg.maxPollinationTicks) { // TenseiMC
                             Bee.this.dropFlower();
                         } else {
                             if (arrivedAtHoverPos) {
@@ -1254,7 +1265,8 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
         }
 
         private Optional<BlockPos> findNearbyFlower() {
-            Iterable<BlockPos> closestNearbyFlowers = BlockPos.withinManhattan(Bee.this.blockPosition(), 5, 5, 5);
+            int radius = Bee.this.level().tensei.mobs.bee.flowerSearchRadius; // TenseiMC
+            Iterable<BlockPos> closestNearbyFlowers = BlockPos.withinManhattan(Bee.this.blockPosition(), radius, radius, radius); // TenseiMC
             Long2LongOpenHashMap tempCache = new Long2LongOpenHashMap();
 
             for (BlockPos pos : closestNearbyFlowers) {
@@ -1317,7 +1329,7 @@ public class Bee extends Animal implements FlyingAnimal, NeutralMob {
 
         private int getWanderThreshold() {
             int distanceReduction = !Bee.this.hasHive() && !Bee.this.hasSavedFlowerPos() ? 16 : 24;
-            return 48 - distanceReduction;
+            return Bee.this.level().tensei.mobs.bee.tooFarDistance - distanceReduction; // TenseiMC
         }
     }
 
