From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 5 May 2026 23:55:40 +0200
Subject: [PATCH] feat: make ChestBlock hardcoded values configurable


diff --git a/net/minecraft/world/level/block/ChestBlock.java b/net/minecraft/world/level/block/ChestBlock.java
index 66608248dfa13aa1a37c168bc1be21017c49fcb5..7f47ea6c2f63858b49f75b23da2c4bfaf21ffb50 100644
--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -253,7 +253,7 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
             type = this.getChestType(context.getLevel(), context.getClickedPos(), facingDirection);
         }
 
-        return this.defaultBlockState().setValue(FACING, facingDirection).setValue(TYPE, type).setValue(WATERLOGGED, replacedFluidState.is(Fluids.WATER));
+        return this.defaultBlockState().setValue(FACING, facingDirection).setValue(TYPE, type).setValue(WATERLOGGED, context.getLevel().tensei.blocks.chest.allowWaterlogging && replacedFluidState.is(Fluids.WATER)); // Tensei - allowWaterlogging
     }
 
     protected ChestType getChestType(final Level level, final BlockPos pos, final Direction facingDirection) {
@@ -408,6 +408,11 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
 
     @Override
     protected int getAnalogOutputSignal(final BlockState state, final Level level, final BlockPos pos, final Direction direction) {
+        // Tensei start - comparatorOutputEnabled
+        if (!level.tensei.blocks.chest.comparatorOutputEnabled) {
+            return 0;
+        }
+        // Tensei end - comparatorOutputEnabled
         return AbstractContainerMenu.getRedstoneSignalFromContainer(getContainer(this, state, level, pos, false));
     }
 
