From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Fri, 8 May 2026 18:45:30 +0200
Subject: [PATCH] feat: make BeehiveBlockEntity hardcoded values configurable


diff --git a/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
index d2aa5ef709781e0cea99883ce8deab340d4adee0..46d564b07a2dd42e4ac617328c92eaab42ce84c0 100644
--- a/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -132,7 +132,7 @@ public class BeehiveBlockEntity extends BlockEntity {
                     if (!this.isSedated()) {
                         bee.setTarget(player, org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_PLAYER); // CraftBukkit
                     } else {
-                        bee.setStayOutOfHiveCountdown(400);
+                        bee.setStayOutOfHiveCountdown(level.tensei.blocks.beehive.minTicksBeforeReenteringHive);
                     }
                 }
             }
@@ -254,7 +254,7 @@ public class BeehiveBlockEntity extends BlockEntity {
             // CraftBukkit end
             if (entity instanceof Bee bee) {
                 RandomSource random = level.getRandom();
-                if (savedFlowerPos != null && !bee.hasSavedFlowerPos() && random.nextFloat() < 0.9F) {
+                if (savedFlowerPos != null && !bee.hasSavedFlowerPos() && random.nextFloat() < level.tensei.blocks.beehive.flowerPosInheritChance) {
                     bee.setSavedFlowerPos(savedFlowerPos);
                 }
 
@@ -263,7 +263,7 @@ public class BeehiveBlockEntity extends BlockEntity {
                     if (state.is(BlockTags.BEEHIVES, s -> s.hasProperty(BeehiveBlock.HONEY_LEVEL))) {
                         int honeyLevel = getHoneyLevel(state);
                         if (honeyLevel < 5) {
-                            int levelIncrease = random.nextInt(100) == 0 ? 2 : 1;
+                            int levelIncrease = random.nextInt(level.tensei.blocks.beehive.doubleHoneyIncreaseChance) == 0 ? 2 : 1;
                             if (honeyLevel + levelIncrease > 5) {
                                 levelIncrease--;
                             }
@@ -336,7 +336,7 @@ public class BeehiveBlockEntity extends BlockEntity {
 
     public static void serverTick(final Level level, final BlockPos blockPos, final BlockState state, final BeehiveBlockEntity entity) {
         tickOccupants(level, blockPos, state, entity.stored, entity.savedFlowerPos);
-        if (!entity.stored.isEmpty() && level.getRandom().nextDouble() < 0.005) {
+        if (!entity.stored.isEmpty() && level.getRandom().nextDouble() < level.tensei.blocks.beehive.workSoundChance) {
             double x = blockPos.getX() + 0.5;
             double y = blockPos.getY();
             double z = blockPos.getZ() + 0.5;
@@ -448,12 +448,24 @@ public class BeehiveBlockEntity extends BlockEntity {
                 BeehiveBlockEntity.IGNORED_BEE_TAGS.forEach(output::discard);
                 CompoundTag entityTag = output.buildResult();
                 boolean hasNectar = entityTag.getBooleanOr("HasNectar", false);
-                return new BeehiveBlockEntity.Occupant(TypedEntityData.of(entity.getType(), entityTag), 0, hasNectar ? 2400 : 600);
+                int minTicks;
+                //noinspection resource
+                if (entity.level() instanceof ServerLevel serverLevel) {
+                    minTicks = hasNectar
+                            ? serverLevel.tensei.blocks.beehive.minOccupationTicksNectar
+                            : serverLevel.tensei.blocks.beehive.minOccupationTicksNectarless;
+                } else {
+                    minTicks = hasNectar ? 2400 : 600; // fallback
+                }
+                return new BeehiveBlockEntity.Occupant(TypedEntityData.of(entity.getType(), entityTag), 0, minTicks);
             }
         }
 
         public static BeehiveBlockEntity.Occupant create(final int ticksInHive) {
-            return new BeehiveBlockEntity.Occupant(TypedEntityData.of(EntityType.BEE, new CompoundTag()), ticksInHive, 600);
+            return create(ticksInHive, 600);
+        }
+        public static BeehiveBlockEntity.Occupant create(final int ticksInHive, final int minTicksInHive) {
+            return new BeehiveBlockEntity.Occupant(TypedEntityData.of(EntityType.BEE, new CompoundTag()), ticksInHive, minTicksInHive);
         }
 
         public @Nullable Entity createEntity(final Level level, final BlockPos hivePos) {
