From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Wed, 6 May 2026 22:22:13 +0200
Subject: [PATCH] feat: Implement BonemealableBlock for SugarCaneBlock


diff --git a/net/minecraft/world/level/block/SugarCaneBlock.java b/net/minecraft/world/level/block/SugarCaneBlock.java
index 8964d69bb1baeefafefb1350f7622eb653697490..cdd9334437d65bb895d4f7533469bfcd8478f348 100644
--- a/net/minecraft/world/level/block/SugarCaneBlock.java
+++ b/net/minecraft/world/level/block/SugarCaneBlock.java
@@ -19,7 +19,7 @@ import net.minecraft.world.level.material.FluidState;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
 
-public class SugarCaneBlock extends Block {
+public class SugarCaneBlock extends Block implements BonemealableBlock {
     public static final MapCodec<SugarCaneBlock> CODEC = simpleCodec(SugarCaneBlock::new);
     public static final IntegerProperty AGE = BlockStateProperties.AGE_15;
     private static final VoxelShape SHAPE = Block.column(12.0, 0.0, 16.0);
@@ -112,4 +112,54 @@ public class SugarCaneBlock extends Block {
     protected void createBlockStateDefinition(final StateDefinition.Builder<Block, BlockState> builder) {
         builder.add(AGE);
     }
+
+    @Override
+    public boolean isValidBonemealTarget(final LevelReader level, final BlockPos pos, final BlockState state) {
+        if (level instanceof ServerLevel serverLevel
+                && !serverLevel.tensei.blocks.sugarCane.bonemealEnabled) {
+            return false;
+        }
+
+        BlockPos topPos = pos;
+        if (level instanceof ServerLevel serverLevel && serverLevel.tensei.blocks.sugarCane.bonemealTargetTop) {
+            while (level.getBlockState(topPos.above()).is(this)) {
+                topPos = topPos.above();
+            }
+        }
+
+        int height = 1;
+        while (level.getBlockState(topPos.below(height)).is(this)) {
+            height++;
+        }
+
+        if (level instanceof ServerLevel serverLevel) {
+            return level.isEmptyBlock(topPos.above())
+                    && height < serverLevel.paperConfig().maxGrowthHeight.reeds;
+        }
+        return level.isEmptyBlock(topPos.above()) && height < 3;
+    }
+
+    @Override
+    public boolean isBonemealSuccess(final net.minecraft.world.level.Level level, final RandomSource random, final BlockPos pos, final BlockState state) {
+        if (level instanceof ServerLevel serverLevel) {
+            return random.nextFloat() < serverLevel.tensei.blocks.sugarCane.bonemealSuccessChance;
+        }
+        return true;
+    }
+
+    @Override
+    public void performBonemeal(final ServerLevel level, final RandomSource random, final BlockPos pos, final BlockState state) {
+        BlockPos topPos = pos;
+        if (level.tensei.blocks.sugarCane.bonemealTargetTop) {
+            while (level.getBlockState(topPos.above()).is(this)) {
+                topPos = topPos.above();
+            }
+        }
+        org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, topPos.above(), this.defaultBlockState(), Block.UPDATE_ALL);
+        level.setBlock(topPos, level.getBlockState(topPos).setValue(AGE, 0), Block.UPDATE_NONE);
+
+        if (level.tensei.blocks.sugarCane.bonemealParticles) {
+            level.sendParticles(net.minecraft.core.particles.ParticleTypes.HAPPY_VILLAGER, topPos.getX() + 0.5, topPos.getY() + 0.5, topPos.getZ() + 0.5, 9, 0.25, 0.25, 0.25, 0.0);
+        }
+    }
 }
