From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Thu, 21 May 2026 15:39:15 +0200
Subject: [PATCH] feat: Add Tensei config support for sugar cane bonemeal


diff --git a/net/minecraft/world/level/block/SugarCaneBlock.java b/net/minecraft/world/level/block/SugarCaneBlock.java
index 31b80cd9173a7fc2e9bfcdcce22beee97f97735b..e815aec7b8ca014d58cd8565412d8e42b718be5c 100644
--- a/net/minecraft/world/level/block/SugarCaneBlock.java
+++ b/net/minecraft/world/level/block/SugarCaneBlock.java
@@ -137,7 +137,10 @@ public class SugarCaneBlock extends Block implements BonemealableBlock { // Canv
         final BlockPos pos,
         final BlockState state
     ) {
-        return level.canvasConfig().farming.sugarCaneBonemealable;
+        if (level.canvasConfig().farming.sugarCaneBonemealable) {
+            return random.nextFloat() < level.tensei.blocks.sugarCane.bonemealSuccessChance;
+        }
+        return false;
     }
 
     @Override
