--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -28,6 +_,12 @@
      */
     Key BRAND_CANVAS_ID = Key.key("canvasmc", "canvas");
     // Canvas end - Rebrand
+    // TenseiMC start - Rebrand
+    /**
+     * The brand id for TenseiMC.
+     */
+    Key BRAND_TENSEI_ID = Key.key("euphyllia", "tensei");
+    // TenseiMC end - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
