--- a/canvas-server/build.gradle.kts
+++ b/canvas-server/build.gradle.kts
@@ -48,7 +_,27 @@
         }
     }
 
-    activeFork = canvas
+     val croissant = forks.register("croissant") {
+         forks = canvas
+         upstream.patchRepo("paperServer") {
+             upstreamRepo = canvas.patchedRepo("paperServer")
+             patchesDir = rootDirectory.dir("croissant-server/paper-patches")
+             outputDir = rootDirectory.dir("paper-server")
+        }
+         upstream.patchRepo("foliaServer") {
+             upstreamRepo = canvas.patchedRepo("foliaServer")
+             patchesDir = rootDirectory.dir("croissant-server/folia-patches")
+             outputDir = rootDirectory.dir("folia-server")
+        }
+        upstream.patchDir("canvasServer") {
+            upstreamPath = "canvas-server"
+            excludes = setOf("src/minecraft", "paper-patches", "folia-patches", "minecraft-patches", "build.gradle.kts", "build.gradle.kts.patch")
+            patchesDir = rootDirectory.dir("croissant-server/canvas-patches")
+            outputDir = rootDirectory.dir("canvas-server")
+        }
+    }
+
+    activeFork = croissant
 
 
     spigot {
@@ -139,12 +_,16 @@
         resources { srcDir("../paper-server/src/main/resources") }
         java { srcDir("../folia-server/src/main/java") }
         resources { srcDir("../folia-server/src/main/resources") }
+        java { srcDir("../canvas-server/src/main/java") }
+        resources { srcDir("../canvas-server/src/main/resources") }
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
         java { srcDir("../folia-server/src/test/java") }
         resources { srcDir("../folia-server/src/test/resources") }
+        java { srcDir("../canvas-server/src/test/java") }
+        resources { srcDir("../canvas-server/src/test/resources") }
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -172,7 +_,7 @@
 }
 
 dependencies {
-    implementation(projects.canvasApi)
+    implementation(projects.croissantApi)
     implementation("ca.spottedleaf:concurrentutil:0.0.8")
     implementation("org.jline:jline-terminal-ffm:3.29.0") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.29.0") // fall back to jni on java 21
@@ -247,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Canvas",
+            "Implementation-Title" to "Croissant",
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Canvas",
+            "Specification-Title" to "Croissant",
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "Canvas Team",
-            "Brand-Id" to "canvasmc:canvas",
-            "Brand-Name" to "Canvas",
+            "Specification-Vendor" to "Euphyllia Team",
+            "Brand-Id" to "canvasmc:croissant",
+            "Brand-Name" to "Croissant",
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
