From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Fri, 8 May 2026 18:57:07 +0200
Subject: [PATCH] feat: make BeehiveBlockEntity hardcoded values configurable


diff --git a/src/main/java/org/bukkit/craftbukkit/block/CapturedBlockState.java b/src/main/java/org/bukkit/craftbukkit/block/CapturedBlockState.java
index efbc098faa4cc2ff99b7f245fb88692314503182..b10dcc42e45a65a007ca86242681d375a80e7169 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CapturedBlockState.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CapturedBlockState.java
@@ -52,12 +52,13 @@ public final class CapturedBlockState extends CraftBlockState {
             BlockPos pos = this.getPosition();
             RandomSource random = level.getRandom();
 
-            // copied from BeehiveDecorator
+            // Begin copied block from BeehiveDecorator
             level.getBlockEntity(pos, BlockEntityType.BEEHIVE).ifPresent(beehive -> {
                 int numBees = 2 + random.nextInt(2);
+                int minTicks = level.getLevel().tensei.blocks.beehive.minOccupationTicksNectarless;
 
                 for (int count = 0; count < numBees; count++) {
-                    beehive.storeBee(BeehiveBlockEntity.Occupant.create(random.nextInt(599)));
+                    beehive.storeBee(BeehiveBlockEntity.Occupant.create(random.nextInt(599), minTicks));
                 }
             });
         }
