From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Toffikk <slodkitofik@gmail.com>
Date: Thu, 24 Jul 2025 21:37:59 +0200
Subject: [PATCH] feature panic


diff --git a/net/minecraft/CrashReport.java b/net/minecraft/CrashReport.java
index ad10eda79e6b76fc1896b4cdb27546b515d1cfa4..b665c762b6d82d6f8d5deb16efebb277d8076f7c 100644
--- a/net/minecraft/CrashReport.java
+++ b/net/minecraft/CrashReport.java
@@ -219,6 +219,6 @@ public class CrashReport {
 
     public static void preload() {
         //MemoryReserve.allocate(); // Paper - Disable memory reserve allocating
-        new CrashReport("Don't panic!", new Throwable()).getFriendlyReport(ReportType.CRASH);
+        new CrashReport("Don't panic!", new Throwable()).getFriendlyReport(ReportType.CRASH); // Croissant - panic.
     }
 }
