--- a/canvas-api/build.gradle.kts
+++ b/canvas-api/build.gradle.kts
@@ -104,20 +_,24 @@
             srcDir(generatedDir)
             srcDir(file("../paper-api/src/main/java"))
             srcDir(file("../folia-api/src/main/java"))
+            srcDir(file("../canvas-api/src/main/java"))
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
             srcDir(file("../folia-api/src/main/resources"))
+            srcDir(file("../canvas-api/src/main/resources"))
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
             srcDir(file("../folia-api/src/test/java"))
+            srcDir(file("../canvas-api/src/test/java"))
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
             srcDir(file("../folia-api/src/test/resources"))
+            srcDir(file("../canvas-api/src/test/resources"))
         }
     }
 }
@@ -152,15 +_,15 @@
             suppressPomMetadataWarningsFor(it)
         }
         from(components["java"])
-        // Canvas start - add more info to the publication
+        // Canvas start - add more info to the publication // TenseiMC start - rebrand
         pom {
             name.set("canvas-api")
-            description.set("API bundle for the Canvas Minecraft server software")
-            url.set("https://github.com/CraftCanvasMC/Canvas")
+            description.set("API bundle for the TenseiMC Minecraft server software")
+            url.set("https://github.com/Euphillya/TenseiMC")
             licenses {
                 license {
                     name.set("GPL-3.0")
-                    url.set("https://github.com/CraftCanvasMC/Canvas/blob/HEAD/LICENSE")
+                    url.set("https://github.com/Euphillya/TenseiMC/blob/HEAD/LICENSE")
                     distribution.set("repo")
                 }
             }
@@ -174,10 +_,10 @@
                 }
             }
             scm {
-                url.set("https://github.com/CraftCanvasMC/Canvas")
+                url.set("https://github.com/Euphillya/TenseiMC")
             }
         }
-        // Canvas end - add more info to the publication
+        // Canvas end - add more info to the publication // TenseiMC end - rebrand
     }
 }
