--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -28,6 +_,12 @@
      */
     Key BRAND_CANVAS_ID = Key.key("canvasmc", "canvas");
     // Canvas end - Rebrand
+    // Croissant start - Rebrand
+    /**
+     * The brand id for Croissant.
+     */
+    Key BRAND_CROISSANT_ID = Key.key("euphyllia", "croissant");
+    // Croissant end - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
